### PR TITLE
ci: run unit and memory tests separately

### PR DESCRIPTION
**Pull request by Westbrook**
## Description
Separate unit and memory tests for CI time efficiencies.

@Rajdeepc @TarunAdobe it looks like we were previously skipping all of the memory tests in CI, because CI runs against non-memory tooling Chrome by default which skips those tests. This move puts the memory tests back into the workflow and four of them seem to be failing. Would yall prefer to work against this branch or to see them skipped herein so you can PR them back on, in a passing state, in individual PRs, later?

I'll also test whether we can run memory tests with concurrency in this setup.

## How has this been tested?
-   [ ] _Test case 1_
    1. [Go to CI](https://app.circleci.com/pipelines/github/adobe/spectrum-web-components/21745/workflows/c6de2dcf-9667-44dc-916d-baf2663a04f7)
    2. See that [test-chromium](https://app.circleci.com/pipelines/github/adobe/spectrum-web-components/21745/workflows/c6de2dcf-9667-44dc-916d-baf2663a04f7/jobs/567560) and [test-chromium-memory](https://app.circleci.com/pipelines/github/adobe/spectrum-web-components/21745/workflows/c6de2dcf-9667-44dc-916d-baf2663a04f7/jobs/567579) are now separate jobs
    3. See that allowing concurrency in Chrome while unit testing gets the job [under 4 minutes](https://app.circleci.com/pipelines/github/adobe/spectrum-web-components/21745/workflows/c6de2dcf-9667-44dc-916d-baf2663a04f7/jobs/567560) with appropriately cached
    4. See that the memory tests have [some failures](https://app.circleci.com/pipelines/github/adobe/spectrum-web-components/21745/workflows/c6de2dcf-9667-44dc-916d-baf2663a04f7/jobs/567579)

## Types of changes
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist
-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [ ] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [ ] I have added tests to cover my changes.
-   [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Rajdeepc** [comment]
@Westbrook I checked these failures and it seems like there can be small memory leaks happening on these elements. Lets skip these 4 from here and we can take it up in separate PRs. I will add 4 internal tickets to track these.

File: web-test-runner.utils.js [hunk 1]
```diff
@@ -47,6 +46,29 @@ export const chromiumWithMemoryTooling = playwrightLauncher({
     },
 });
 
+export const chromiumWithMemoryToolingCI = playwrightLauncher({
+    product: 'chromium',
+    concurrency: 1,
```
Comment (Westbrook) [inline web-test-runner.utils.js:51]: @Rajdeepc while you're working through the new VRT failures here, can you test whether the separation of unit and memory tests allows us to leverage concurrency to get the memory test into the same general time scale as the unit/list/preview tasks? If we could get this one under 5 minutes as well, that would be great!

**Comment by Rajdeepc** [comment]
Adding `ifDefined` directive to the style attributes fixes these errors. Every time the element is rendered, new string is getting created for the style attribute, even if the value has not changed. This is leading to memory leaks.

**Comment by Westbrook** [comment]
> Adding `ifDefined` directive to the style attributes fixes these errors. Every time the element is rendered, new string is getting created for the style attribute, even if the value has not changed or assigned. This is leading to memory leaks.

Doesn't seem that it _ever_ creates the style string in that context: https://separate-memory-and-unit-testing--spectrum-web-components.netlify.app/storybook/?path=/story/color-area--default Would it be better to cache the string or to use a `styleMap()` in attempt to deliver the component correct and yet not leak?

**Comment by Rajdeepc** [comment]
> > Adding `ifDefined` directive to the style attributes fixes these errors. Every time the element is rendered, new string is getting created for the style attribute, even if the value has not changed or assigned. This is leading to memory leaks.
> 
> Doesn't seem that it _ever_ creates the style string in that context: https://separate-memory-and-unit-testing--spectrum-web-components.netlify.app/storybook/?path=/story/color-area--default Would it be better to cache the string or to use a `styleMap()` in attempt to deliver the component correct and yet not leak?

Adding a `styleMap` not caching or abstracting the value rather I found out that it is leaking some unrelated memory which is a thing to check. But to make up for the lost VRTs, `ifDefined` has been replaced with `styleMap`

**Comment by Rajdeepc** [comment]
@Westbrook I have passed all the tests, fixing the Swatch memory leak. Will take up `color-area, color-wheel and color-slider` in a separate PR and unblock this going to `main`. Feel free to add your thoughts so that we can land this!

File: web-test-runner.utils.js (hunk 1 above)
Comment (Rajdeepc) [reply web-test-runner.utils.js:51]: Screenshot 2024-05-10 at 4 27 52 PM

**Review by Rajdeepc — APPROVED** [review]

**Pull request merged**